File: src/main/java/org/forif/subwayapp/controller/UserController.java
package org.forif.subwayapp.controller;

import lombok.AllArgsConstructor;
import org.forif.subwayapp.entity.SubwayUser;
import org.forif.subwayapp.service.UserService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/users")
@AllArgsConstructor
public class UserController {
    private final UserService userService;

    @GetMapping({"/{id}"})
    public ResponseEntity<SubwayUser> getUserById(
            @PathVariable Long id
    ) {
        return ResponseEntity.status(200).body(userService.getUserById(id));
    }
}

File: subway-app/src/main/kotlin/com/subway/controller/UserController.kt
package com.subway.controller

import com.subway.dto.UserDTO
import com.subway.service.UserService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("api/v1/users")
class UserController(private val userService: UserService) {

    @GetMapping("/{id}")
    fun getUserById(@PathVariable id: Long): ResponseEntity<UserDTO> {
        val user = userService.getUserById(id)
        return ResponseEntity.ok(user)
    }

    @PostMapping
    fun createUser(@RequestBody userDTO: UserDTO): ResponseEntity<UserDTO> {
        val createdUser = userService.createUser(userDTO)
        return ResponseEntity.status(HttpStatus.CREATED).body(createdUser)
    }

    @DeleteMapping("{id}")
    fun deleteUserById(@PathVariable id: Long){
        return userService.deleteUserById(id)
    }


}